examples/examples/data-channels-flow-control/data-channels-flow-control.rs

```diff
@@ -1,368 +1,263 @@
-            //urls: vec!["stun:stun.l.google.com:19302".to_owned()],
-            ..Default::default()
-        }],
-        ..Default::default()
-    };
+                    let task = tokio::spawn(async move {
+                        let start = SystemTime::now();
 
-    // Create a new RTCPeerConnection
-    let pc = Arc::new(api.new_peer_connection(config).await?);
+                        tokio::time::sleep(Duration::from_secs(1)).await;
+                        println!("");
 
-    pc.on_data_channel(Box::new(move |dc: Arc<RTCDataChannel>| {
-        let total_bytes_received = Arc::new(AtomicUsize::new(0));
+                        loop {
+                            let total_bytes = get_total_bytes_received.load(Ordering::Relaxed);
 
-        Box::pin(async move {
-            // Register channel opening handling
-            let total_bytes_received2 = Arc::clone(&total_bytes_received);
-            dc.on_open(Box::new(move || {
-                println!("OnOpen: Start receiving data"); //, dc2.label(), dc2.id());
-                tokio::spawn(async move {
-                    let since = SystemTime::now();
+                            let elapsed = SystemTime::now().duration_since(start);
+                            let bps = (total_bytes * 8) as f64 / elapsed.unwrap().as_secs_f64();
 
-                    // Start printing out the observed throughput
-                    loop {
-                        tokio::time::sleep(Duration::from_secs(1)).await;
-                        let total_bytes = total_bytes_received2.load(Ordering::SeqCst);
-                        let bps = (total_bytes * 8) as f64
-                            / SystemTime::now()
-                                .duration_since(since)
-                                .unwrap()
-                                .as_secs_f64();
-                        println!("Throughput: {:.03} Mbps", bps / (1024 * 1024) as f64);
-                    }
-                });
+                            println!(
+                                "Throughput is about {:.03} Mbps",
+                                bps / (1024 * 1024) as f64
+                            );
+                            tokio::time::sleep(Duration::from_secs(1)).await;
+                        }
+                    });
 
-                Box::pin(async {})
+                    task.await.unwrap();
+                })
             }))
             .await;
 
-            // Register the OnMessage to handle incoming messages
-            dc.on_message(Box::new(move |msg: DataChannelMessage| {
-                let n = msg.data.len();
-                total_bytes_received.fetch_add(n, Ordering::SeqCst);
-                Box::pin(async {})
+            dc.on_message(Box::new(move |msg| {
+                let total_bytes_received = total_bytes_received.clone();
+
+                Box::pin(async move {
+                    total_bytes_received.fetch_add(msg.data.len(), Ordering::Relaxed);
```

Comment:
I'm not super clued in to the atomic Orderings, but I would like to continue using the strong ordering of SeqCst.

Reply:
There is no need for this, it's just a counter that only increases, Ordering::Relaxed is the best Ordering in this case.

Reply:
I didn't know that, thanks for teaching me :)